tests/unit/amo/components/TestAddon.js

```diff
@@ -748,6 +771,242 @@ describe('Addon', () => {
+
+      sinon.assert.calledWith(fakeDispatch, fetchOtherAddonsByAuthors({
+        addonType: addon.type,
+        authors: addon.authors.map((author) => author.username),
+        errorHandlerId: createStubErrorHandler().id,
+        slug: addon.slug,
+      }));
+    });
+
+    it('does not fetch the other add-ons when add-on is the same', () => {
+      const addon = fakeAddon;
+      const fakeDispatch = sinon.stub();
+      const root = shallowRender({ addon, dispatch: fakeDispatch });
+      fakeDispatch.reset();
+
+      root.setProps({ addon });
+
+      sinon.assert.notCalled(fakeDispatch);
+    });
+
+    it('fetches the other add-ons when a new add-on is loaded', () => {
+      const fakeDispatch = sinon.stub();
+      const root = shallowRender({ addon: fakeAddon, dispatch: fakeDispatch });
+      fakeDispatch.reset();
+
+      const newAddon = { ...fakeAddon, slug: 'new-addon-slug' };
+      root.setProps({ addon: newAddon });
+
+      sinon.assert.calledWith(fakeDispatch, fetchOtherAddonsByAuthors({
+        addonType: newAddon.type,
+        authors: newAddon.authors.map((author) => author.username),
+        errorHandlerId: createStubErrorHandler().id,
+        slug: newAddon.slug,
+      }));
+    });
+
+    it('does not fetch the other add-ons when they are already loaded', () => {
+      const fakeDispatch = sinon.stub();
+      const root = shallowRender({ addon: fakeAddon, dispatch: fakeDispatch });
+      fakeDispatch.reset();
+
+      const newAddon = { ...fakeAddon, slug: 'new-addon-slug' };
+      // These `addonsByAuthors` are related to the `newAddon` above, and it
+      // simulates the case when user has previously browsed this `newAddon`
+      // and navigates to it again.
+      const addonsByAuthors = [
+        { ...fakeAddon, slug: 'addon-1' },
+        { ...fakeAddon, slug: 'addon-2' },
+        { ...fakeAddon, slug: 'addon-3' },
+      ];
+
+      root.setProps({ addon: newAddon, addonsByAuthors });
+
+      sinon.assert.notCalled(fakeDispatch);
+    });
+
+    it('is hidden when an add-on has not loaded yet', () => {
+      const root = shallowRender({ addon: undefined });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveLength(0);
+    });
+
+    it('is hidden when other addons are not loaded yet', () => {
+      const root = shallowRender({ addonsByAuthors: undefined });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveLength(0);
+    });
+
+    it('is hidden when there is no other addon', () => {
+      const root = shallowRender({ addonsByAuthors: [] });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveLength(0);
+    });
+
+    it('displays the developer name when add-on is an extension', () => {
+      const root = mountMoreAddonsByAuthors({
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.Card-header'))
+        .toIncludeText('More extensions by Krupa');
```

Comment:
The i18n components are a little tricky to test but you don't need `mount()`. You can still do it with `shallow()` by checking the HTML:

```js
expect(root.find('.Card-header').html())
  .toContain('More extensions by Krupa');
```

If you need to check for links or something you could do it like this:

```js
expect(root.find('.Card-header').render().find('a'))
  .toHaveLength(1);
```

Reply:
I don't see how to do that yet since switching to `mount` gives these errors:

```
  ● Addon › more add-ons by authors › indicates when other add-ons are themes

    TypeError: Cannot read property 'sprintf' of undefined

      at SearchResultBase.renderResult (src/amo/components/SearchResult/index.js:114:27)
      at SearchResultBase.render (src/amo/components/SearchResult/index.js:127:25)
```

Reply:
You'll need to use:

```js
import { getFakeI18nInst } from 'tests/unit/helpers';

shallowUntilTarget(<MyComponent i18n={getFakeI18nInst()} />, MyComponentBase);
```

Reply:
I'll give it a try but I thought that was already there: https://github.com/willdurand/addons-frontend/blob/0ddc2d7368430cd8a7fcd48e4cd82cdc53b3c0cd/tests/unit/amo/components/TestAddon.js#L105-L108. Does `shallowUntilTarget` do something different?

Reply:
`shallowUntilTarget` will use the wrapped component which will automatically add in the `i18n` to the props via the `compose(translate(), [...])` wrapping the component.

Reply:
So come to think of it you might not even need to add in the fakeI18n thing, but I forget 😆